packages/nx-plugin/src/generators/playwright-component-configuration/generator.ts

```diff
@@ -0,0 +1,38 @@
+import {
+  getProjects,
+  formatFiles,
+  generateFiles,
+  Tree,
+  updateProjectConfiguration,
+} from '@nx/devkit';
+import * as path from 'path';
+import { PlaywrightComponentConfigurationGeneratorSchema } from './schema';
+
+export default async function (
+  tree: Tree,
+  options: PlaywrightComponentConfigurationGeneratorSchema
+) {
+  const { name } = options;
+  const projects = getProjects(tree);
+  const project = projects.get(name);
+  if (!project) {
+    return;
+  }
+
+  const { root: projectRoot, targets } = project;
+  if (!targets) {
+    return;
+  }
+
+  targets['component-test'] = {
+    executor: '@fluentui-contrib/nx-plugin:playwright',
+    options: {
+      testingType: 'component',
+    },
+  };
+
+  updateProjectConfiguration(tree, name, project);
+
+  generateFiles(tree, path.join(__dirname, 'files'), projectRoot, options);
```

Comment:
there are more explicit APIs to update project targets
 
```suggestion
  const config = readProjectConfiguration(tree, name);

  const targetDefinition = {
    executor: '@fluentui-contrib/nx-plugin:playwright',
    options: {
      testingType: 'component',
    },
  };

  config.targets = config.targets ?? {};
  config.targets['component-test'] = targetDefinition;

  updateProjectConfiguration(tree, name, { ...config });

  generateFiles(tree, path.join(__dirname, 'files'), config.root, options);
```

Reply:
Thx, I'll update it accordingly. I was following the pattern I've seen in the storybook configuration, might be nice to update there also in a separate PR

https://github.com/microsoft/fluentui-contrib/blob/3de28ae150fb8a004eafa09f20e62740f8d84dde/packages/nx-plugin/src/generators/configure-storybook/generator.ts#L10-L26